### PR TITLE
WELD-2774 Implement BM methods for event/bean assignability

### PR DESCRIPTION
**Pull request by manovotn**
The PR is going to fail CI until we get a CDI API release and ideally a TCK release to have tests.

There are no Weld specific tests for this as Weld doesn't add anything beyond what's in the TCK coverage via https://github.com/jakartaee/cdi-tck/pull/526

### CODE REVIEW

**Comment by manovotn** [comment]
I've added some validations:
* Qualifier validation for all inputs (if the annotation is a CDI qualifier)
* Event type validation for unresolved type variable
* Filtering of provided bean types for legal bean types only

These are so far not covered by the linked TCK test.
If they won't be covered, I'll add Weld test cases.

File: impl/src/main/java/org/jboss/weld/manager/BeanManagerImpl.java [hunk 1]
```diff
@@ -1563,6 +1568,73 @@ public WeldInstance<Object> createInstance() {
...
+        validateQualifiers(observedEventQualifiers, loggingMethodName);
+        // type check first
+        if (EventTypeAssignabilityRules.instance().matches(observedEventType,
+                HierarchyDiscovery.forNormalizedType(eventType).getTypeClosure())) {
+            // normalize qualifiers, adding built-ins where missing, then compare
+            Set<Annotation> normalizedEventQualifiers = new HashSet<>(eventQualifiers);
+            if (eventQualifiers.isEmpty()
+                    || (eventQualifiers.size() == 1 && eventQualifiers.iterator().next().equals(Any.Literal.INSTANCE))) {
+                normalizedEventQualifiers.add(Default.Literal.INSTANCE);
+            }
```
Comment (Ladicek) [inline impl/src/main/java/org/jboss/weld/manager/BeanManagerImpl.java:1619]: With our recently acquired deep knowledge of the subject, should this be just `if (eventQualifiers.isEmpty())`?

Comment (manovotn) [reply impl/src/main/java/org/jboss/weld/manager/BeanManagerImpl.java:1619]: Yes, I believe it should.

Comment (manovotn) [reply impl/src/main/java/org/jboss/weld/manager/BeanManagerImpl.java]: This should now be fixed

**Comment by manovotn** [comment]
Superseeded by https://github.com/weld/core/pull/2911

**Pull request closed without merging**